awsTerminate

```diff
@@ -3,9 +3,31 @@ import sys
 import argparse
 import aws_kpc, aws_icc
 
-parser = argparse.ArgumentParser()
+
+
+
+inList = []
+
+def terminateInstances():
+    if inList:
+        aws_icc.aws_terminate_instance(inList)
+        for instance in inList:
+            stateCode,stateString = aws_icc.aws_instance_state(instance)
+            if stateCode == 32 or stateCode == 48:
+                # State 32 indicates the instance is shutting-down
+                # State 48 indicates the instance is terminated
+                print('Instance {}: {}'.format(instance,stateString))
+            else:
+                print('Received unexpected state ({}) {} when terminateing instance{}'.format(stateCode,stateString,instance))
+
+
+# Work starts here
+parser = argparse.ArgumentParser(description='Shutdown one or more running instances')
 parser.add_argument("-i","--id", help="The instance id to terminate.",type=str)
-# parser.add_argument("-a","--all", help="Terminate all running instances.")
+parser.add_argument("-a","--all", help="Termanate all running instances.",action="store_true")
+if len(sys.argv) < 2:
```

Comment:
Will this print usage when given a bad flag, or only when the required argument is missing?

Reply:
In the current code state, it will provide info with a bad flag:

Phileas:awsWorks TED$ ./awsTerminate -p
usage: awsTerminate [-h] [-i ID] [-a]
awsTerminate: error: unrecognized arguments: -p

or if any flag is missing:

Phileas:awsWorks TED$ ./awsTerminate
usage: awsTerminate [-h] [-i ID] [-a]
Phileas:awsWorks TED$ ./awsTerminate -i
usage: awsTerminate [-h] [-i ID] [-a]
awsTerminate: error: argument -i/--id: expected one argument
Phileas:awsWorks TED$ ./awsTerminate -a
Nothing to terminate.